### PR TITLE
Make aura dependencies move only one way

### PR DESCRIPTION
**Pull request by asdacap**
- Some dependencies in AuRA is added later instead of via constructor.
- This makes it very hard to follow.

## Changes

- Pass `IAuraValidator` and `IReportingValidator` via constructor.

#### What types of changes does your code introduce?

- [X] Refactoring

#### Requires testing

- [X] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [X] No

#### Notes on testing

- No idea how to test this.

### CODE REVIEW

**Review by benaadams — APPROVED** [review]

**Review by LukaszRozmej — CHANGES_REQUESTED** [review]
AuRa contract validator needs to modify state, which if I understand won't be able to do now?
Can you try full syncing some larger parts of aura chains?

**Comment by asdacap** [comment]
Nope, WidthrawalManager is needed by auraValidator.
